check/test.go

```diff
@@ -220,13 +233,19 @@ func (ts *tests) execute(s string) *testOutput {
 	case and, "":
 		result = true
 		for i := range res {
+			finalOutput.ExpectedResult += fmt.Sprintf("%s AND ", res[i].ExpectedResult)
```

Comment:
Use [strings.Join()](https://golang.org/pkg/strings/#Join) for these, it's neater and means you don't have to manually delete the last iteration

Reply:
You are right, but this for loop will still happen so it makes it not efficient since we're iterating on the array 2 times

Reply:
I can go with whatever you prefer, what do you say?

Reply:
strings.Join() is (a) idiomatic (b) [almost certainly more efficient than roll-your-own string concatenations with fmt.Printf](https://syslog.ravelin.com/bytes-buffer-i-thought-you-were-my-friend-4148fd001229).

Reply:
Wow amazing! you really blow my mind right now :)
So the only problem I see now is that I need to join not the res array but the ExpectedResult property of each item in that array.
What I can do now is either keep this code or maybe append the ExpectedResult of each item to a new array, and then call the strings.Join() method.
What do you think?

Reply:
Good point - I'd probably do it something like [this](https://play.golang.org/p/8uh_FW1MK53)

Reply:
Done